VSTSDropProxy.cs

```diff
@@ -83,141 +85,106 @@ public VSTSDropProxy(string VSTSDropUri, string path, string pat, TimeSpan blobt
         /// <returns>The manifest uri.</returns>
         private static Uri Munge(Uri vstsDropUri, string apiVersion)
         {
-            var querystring = HttpUtility.ParseQueryString(vstsDropUri.Query);
             string manifestpath = vstsDropUri.AbsolutePath.Replace("_apis/drop/drops", "_apis/drop/manifests");
             var uriBuilder = new UriBuilder(vstsDropUri.Scheme, vstsDropUri.Host, -1, manifestpath);
             var queryParameters = HttpUtility.ParseQueryString(uriBuilder.Query);
-            queryParameters.Add(RestfulDropApi.APIVersionParam,apiVersion);
+            queryParameters.Add(RestfulDropApi.APIVersionParam, apiVersion);
             uriBuilder.Query = queryParameters.ToString();
-            
             return uriBuilder.Uri;
         }
 
         private async Task Download(string sasurl, string localpath)
         {
             await Policy
                 .Handle<HttpRequestException>()
+                .Or<SocketException>()
                 .Or<IOException>()
                 .Or<TaskCanceledException>()
-                .WaitAndRetryAsync(5, 
+                .WaitAndRetryAsync(10,
                     retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
-                    (e,t) => 
+                    (e, t) =>
                     {
                         Console.WriteLine($"Exception {e} on {sasurl} -> {localpath}");
-                        File.Delete(localpath);
+                        if (File.Exists(localpath))
+                        {
+                            File.Delete(localpath);
+                        }
                     })
                 .ExecuteAsync(async () =>
                 {
-                    //TODO: Would be better to  set the timeout based on the size of the blob which is in the manifest
```

Comment:
leave my todo?

Reply:
and the 1-5 comments below.

Reply:
I added it back after implementing number 3. I understand what 1 is talking about (if the sub dir specifier isn't enough), but the other ones aren't clear to me.